src/tensors/tensor.jl

```diff
@@ -315,11 +316,14 @@ end
 
 for randf in (:rand, :randn, :randexp, :randisometry)
     _docstr = """
-        $randf([rng=default_rng()], [T=Float64], codomain::ProductSpace{S,N₁},
+        $randf([rng=default_rng()], [TorA=Float64], codomain::ProductSpace{S,N₁},
                      domain::ProductSpace{S,N₂}) where {S,N₁,N₂,T} -> t
-        $randf([rng=default_rng()], [T=Float64], codomain ← domain) -> t
+        $randf([rng=default_rng()], [TorA=Float64], codomain ← domain) -> t
         
     Generate a tensor `t` with entries generated by `$randf`.
+    The type `TorA` can be used to control the element type and
+    data type generated. For example, if `TorA` is a `SparseVector{ComplexF32}`,
+    then the final output `TensorMap` will have that as its storage type.
```

Comment:
That might not be the best example, as we have `A<:DenseVector` in the `TensorMap` type. We might have to reconsider this at some point, but many implementations do implicitly assume this.